### DISCUSSION TITLE
Is there any way for a @field_serializer method to indicate the field should be omitted entirely?

### DISCUSSION THREAD

**Discussion by bjmc** [pydantic/pydantic#9703] [Question]
I'm imagining something like this:

```python
@field_serializer('x', 'y')
def omit_zeroes(value):
    if value == 0:
        return PydanticUndefined
    return value
```

I tried returning `None` in combination with `model_dump(exclude_none=True)` but it doesn't appear to work - I'm guessing because the custom serializer fires after first filtering None valued-fields.

Does anyone have a good solution for this? I suppose I could write a `@model_serializer` but it seems overkill.

**Comment by aminabbasov** [comment]
bump pls
